web/html/src/manager/virtualization/guests.js

```diff
@@ -0,0 +1,409 @@
+const {Table, Column, SearchField, Highlight} = require("../../components/table");
+const Network = require("../../utils/network");
+const Functions = require("../../utils/functions");
+const Utils = Functions.Utils;
+const {LinkButton, AsyncButton} = require("../../components/buttons");
+const {DeleteDialog} = require("../../components/dialog/DeleteDialog");
+const {DangerDialog} = require("../../components/dialog/DangerDialog");
+const {ModalButton} = require("../../components/dialog/ModalButton");
+const Systems = require("../../components/systems");
+
+class VirtualGuests extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    ['refreshServerData', 'handleSelectItems', 'selectGuest', 'action', 'onBeforeUnload']
+          .forEach(method => this[method] = this[method].bind(this));
+
+    this.state = {
+      actionsResults: {},
+      serverData: null,
+      errors: [],
+      selectedItems: []
+    }
+  }
+
+  onBeforeUnload(event) {
+      this.setState({
+          pageUnloading: true
+      });
+  }
+
+  componentWillMount() {
+    this.refreshServerData();
+    setInterval(this.refreshServerData, this.props.refreshInterval);
+  }
+
+  componentDidMount() {
+    if (salt_entitled) {
+      var port = window.location.port;
+      var url = "wss://" +
+        window.location.hostname +
+        (port ? ":" + port : "") +
+         "/rhn/websocket/minion/virt-notifications";
+      var ws = new WebSocket(url);
+
+      ws.onopen = () => {
+        // Tell the websocket that we want to hear from all action results on this virtual host.
+        ws.send(server_id);
+      }
+
+      ws.onclose = (e) => {
+        var errs = this.state.errors ? this.state.errors : [];
+        if (!this.state.pageUnloading && !this.state.websocketErr) {
+            errs.push(t("Websocket connection closed. Refresh the page to try again."));
+        }
+        this.setState({
+            errors: errs,
+        });
+      };
+
+      ws.onerror = (e) => {
+        console.log("Websocket error: " + e);
+        this.setState({
+           errors: [t("Error connecting to server. Refresh the page to try again.")],
+           websocketErr: true
+        });
+      };
+
+      ws.onmessage = (e) => {
+        var event = JSON.parse(e.data);
+
+        var actions = this.state.actionsResults
+        var uuid = Object.keys(actions).find(key => actions[key].id == event.action);
+        if (uuid != undefined) {
+          actions[uuid].result = event.status;
+          this.setState({
+            actionsResults: actions
+          });
+        }
+      }
+
+      window.addEventListener("beforeunload", this.onBeforeUnload)
+
+      this.setState({
+        websocket: ws
+      });
+    }
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("beforeunload", this.onBeforeUnload)
+  }
+
+  refreshServerData() {
+    Network.get(`/rhn/manager/api/systems/details/virtualization/guests/${server_id}/data`, "application/json").promise
```

Comment:
Lets decouple all the api calls for their own component!! As I said on the comment before, look at this example: https://github.com/renner/uyuni/commit/162057d9f31b78db7f6ced5833a978e0af77517e#diff-20041f7e91b6a047ebef7f75aa7407bcR4

Ideally we would have a folder core/api/bunch_APIcomponents_decoupled_from_render.*.js 

This would be awesome for the plugins/core system we want to implement. For me to develop a new plugin  you wouldn't need to implement any logic of this kind but only connect "core api components" with "core interface components"

```javascript
    <random_api>
         {
              ({data, isloading}) => (
                    <component1/>
                    <table data=data></table>
                    dashboard plugin xpto
                    <component2>
              )
         }
    </random_api>
```